plugin-server/src/main/ingestion-queues/session-recording/utils.ts

```diff
@@ -257,7 +257,7 @@ export const parseKafkaMessage = async (
         return dropMessage('invalid_json', { error, team_id: teamIdWithConfig.teamId })
     }
 
-    const { $snapshot_items, $session_id, $window_id, $snapshot_source } = event.properties || {}
+    const { $snapshot_items, $session_id, $window_id, $snapshot_source, $snapshot_library } = event.properties || {}
```

Comment:
I think `$snapshot_library` should just come from `$lib` which the Mobile SDKs already do, instead of introducing a new flag with a duplicated value, this was discussed in a slack thread and @zlwaterfield preferred to not dupe this value

Reply:
> I think $snapshot_library should just come from $lib which the Mobile SDKs already do

I didn't think we have $lib at this point in the code
web definitely doesn't send it

Reply:
can you share a mobile payload?
do all the mobile SDKs sned the same values?

Reply:
no, we agreed to add those as part of replays a while ago and I added them, so when we needed them, the SDKs were already sending them:

$lib: 'posthog-react-native'
$lib: 'posthog-ios'
$lib: 'posthog-flutter'
$lib: 'posthog-android'
$lib_version: '...the respective version of each sdk'

Reply:
Ah, cool... 🙌 

So

* teach capture to forward lib and lib version when available
* in capture if lib and lib version are not available if user agent includes posthog then use user agent as lib
* in capture if lib and lib version are not available if user agent does not include posthog then == web
* in capture if lib and lib version are available then snapshot lib = lib + lib version 
    * version increases cardinality but makes other querying/investigation easier
* in blob ingestion make sure we're forwarding snapshot lib
* use the info in billing
* write queries to use the info when it is present since in 90 days almost all recordings will have a sensible value stored

Reply:
https://github.com/PostHog/posthog-android/pull/176#discussion_r1758724529